extensions/azure/cosmos-common/src/main/java/org/eclipse/dataspaceconnector/cosmos/azure/CosmosDbApiImpl.java

```diff
@@ -0,0 +1,130 @@
+    }
+
+    @Override
+    public void createItem(Object item) {
+        System.out.println("Hello Ben");
+        try {
+            CosmosItemResponse<Object> response = container.createItem(item, partitionKey, itemRequestOptions);
+            handleResponse(response);
+        } catch (CosmosException e) {
+            throw new EdcException(e);
+        }
+    }
+
+    @Override
+    public @Nullable Object queryItemById(String id) {
+        CosmosItemResponse<Object> response;
+        try {
+            response = container.readItem(id, partitionKey, itemRequestOptions, Object.class);
+        } catch (NotFoundException e) {
+            return null;
+        } catch (CosmosException e) {
+            throw new EdcException(e);
+        }
+        handleResponse(response);
+        return response.getItem();
+    }
+
+    @Override
+    public List<Object> queryAllItems() {
+        try {
+            return container.readAllItems(partitionKey, queryRequestOptions, Object.class).stream().collect(Collectors.toList());
+        } catch (CosmosException e) {
+            throw new EdcException(e);
+        }
+    }
+
+    @Override
+    public List<Object> queryItems(SqlQuerySpec query) {
+        try {
+            return container.queryItems(query, queryRequestOptions, Object.class).stream().collect(Collectors.toList());
+        } catch (CosmosException e) {
+            throw new EdcException(e);
+        }
+    }
+
+    private static void handleResponse(CosmosItemResponse<?> response) {
+        int code = response.getStatusCode();
+        if (code < 200 || code >= 300) {
+            throw new EdcException("Error during CosmosDB interaction: " + code);
+        }
+    }
+
+    private static CosmosContainer getContainer(Vault vault, AbstractCosmosConfig config) {
+        CosmosClient client = createClient(vault, config.getAccountName(), Collections.singletonList(config.getPreferredRegion()));
+        CosmosDatabase database = getDatabase(client, config.getDbName());
+        if (database.readAllContainers().stream().noneMatch(sp -> sp.getId().equals(config.getContainerName()))) {
+            throw new EdcException("No CosmosDB container named '" + config.getContainerName() + "' was found in account '" + config.getAccountName() + "'. Please create one, preferably using terraform.");
+        }
+        return database.getContainer(config.getContainerName());
+    }
+
+    private static CosmosClient createClient(Vault vault, String accountName, List<String> preferredRegions) {
+        var accountKey = vault.resolveSecret(accountName);
+        if (StringUtils.isNullOrEmpty(accountKey)) {
+            throw new EdcException("No credentials found in vault for Cosmos DB '" + accountName + "'");
+        }
+
+        // create cosmos db api client
+        String host = String.format("%s://%s.%s", CLIENT_PROTOCOL, accountName, CLIENT_URL_SUFFIX);
```

Comment:
sorry if I misspoke, what I actually meant was a constant `private final static string HOST_TEMPLATE="https://%s.documents.azure.com:443/"` and then here you simply write `String host = String.format(HOST_TEMPLATE, accountName)`